.azure-pipelines/build-template.yml

```diff
@@ -186,7 +186,14 @@ jobs:
         make check
         gcovr --version
         find SAI/meta -name "*.gc*" | xargs rm -vf
-        gcovr -r ./ -e ".*/SAI/.*" -e ".+/json.hpp" -e "swss/.+" -e ".*/.libs/.*" -e ".*/debian/.*" --exclude-unreachable-branches -x --xml-pretty -o coverage.xml
+        gcov_dirs=$(find . -path "*.libs*gcda" | xargs dirname | sort -u |  cut -c"3-")
+        for dir in ${gcov_dirs}; do
+          source_dir=$(dirname $dir)
+          output_file="coverage-$source_dir.json"
+          gcovr --json-pretty -o $output_file --object-directory $source_dir $dir
```

Comment:
Do you need to add in `--exclude-unreachable-branches` here?

Reply:
good catch, added